### PR TITLE
feat: added retainEmptyParents options (defaults to true)

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -10,10 +10,52 @@ describe("Testing Rewrite", () => {
...
+
+    it("Testing Retain Empty List (Child Match), retainEmptyParents = false", () => {
+      expect(execTest(["test.id"], false)).to.deep.equal({});
+    });
+
+    it("Testing Retain Empty List (No Match), retainEmptyParents = true", () => {
+      expect(execTest(["unknown"], true)).to.deep.equal({});
+    });
+
+    it("Testing Retain Empty List (No Match), retainEmptyParents = false", () => {
```
Comment (MacMcIrish) [inline test/index.spec.js]: Pretty verbose 'it' names that match the parent 'describe', maybe unnecessary?

Comment (simlu) [reply test/index.spec.js]: The problem is the naming collision complains... Not sure what best practises are here? Any idea?

Comment (MacMcIrish) [reply test/index.spec.js]: If 2 `it`s have the same name across multiple `describe`s they have naming collisions? That seems like goofy behaviour. Not a lot you can do about that though...

File: src/index.js [hunk 2]
```diff
@@ -43,6 +44,12 @@ module.exports = ({
       if (retain.includes(needle)) {
         toRetain.push(key);
       }
```
[the remark this answers is not in the document]
Comment (simlu) [reply src/index.js]: Could be, but I do like this explicit

File: test/index.spec.js (hunk 1 above)
Comment (simlu) [reply test/index.spec.js]: Yeah, that's unfortunately how it would seem...

**Pull request merged**